solutions/wilson_romero/solution.py

```diff
@@ -7,8 +7,13 @@ class Solution:
 
     def duplicate_zeros(self, arr: List[int]):
         """Duplicate zeros and shift right"""
-        index = 0
+        length = len(arr)
+        zeros = []
         for step, value in enumerate(arr):
-            if value == 0 and step >= index:
-                arr[step + 1:] = arr[step:-1]
-                index = step + 2
+            if value == 0:
+                zeros.append(step)
```

Comment:
This is creating additional space, i.e.: for an array full of `0`s of length **n**, you'll be creating an auxiliary array of size n containing its indices.